atest/testdata/running/continue_on_failure_tag.robot

```diff
@@ -0,0 +1,265 @@
+    ...    4) 4
+
+    [Tags]   robot:continue-on-failure
+    IF   1==1
+        Fail    1
+        Fail    2
+    END
+    IF   1==2
+        No Operation
+    ELSE
+        Fail    3
+        Fail    4
+    END
+
+Continue in IF with set and remove tag
+    [Documentation]    FAIL ${HEADER}\n\n
+    ...    1) 1\n\n
+    ...    2) 2\n\n
+    ...    3) 3
+
+    Set Tags   robot:continue-on-failure
+    IF   1==1
+        Fail    1
+        Fail    2
+    END
+    Remove Tags   robot:continue-on-failure
+    IF   1==2
+        No Operation
+    ELSE
+        Fail    3
+        Fail    this is not executed
+    END
+
+Continue in IF without tag
+    [Documentation]    FAIL 1
+    IF   1==1
+        Fail    1
+        Fail    This should not be executed
+    END
+
+Continue in IF in UK with tag
+    [Documentation]    FAIL ${HEADER}\n\n
+    ...    1) kw1a\n\n
+    ...    2) kw1b\n\n
+    ...    3) kw1c\n\n
+    ...    4) kw1d
+    If in user keyword with tag
+
+Continue in IF in UK without tag
+    [Documentation]    FAIL kw1a
+    If in user keyword without tag
+
+Run Keywords with tag
+    [Documentation]    FAIL ${HEADER}\n\n
+    ...    1) 1\n\n
+    ...    2) 2
+    [Tags]   robot:continue-on-failure
+    Run Keywords    Fail   1   AND   Fail   2
+
+Recursive Continue in test with tag and two nested UK without tag
+    [Documentation]    FAIL ${HEADER}\n\n
+    ...    1) kw2a\n\n
+    ...    2) kw2b\n\n
+    ...    3) kw2a\n\n
+    ...    4) kw2b\n\n
+    ...    5) This should be executed
+    [Tags]   robot:continue-on-failure-recursive
+    Failure in user keyword without tag     run_kw=Failure in user keyword without tag
+    Fail   This should be executed
+
+Recursive Continue in test with set tag and two nested UK without tag
+    [Documentation]    FAIL ${HEADER}\n\n
+    ...    1) kw2a\n\n
+    ...    2) kw2b\n\n
+    ...    3) kw2a\n\n
+    ...    4) kw2b\n\n
+    ...    5) This should be executed
+    Set Tags   robot:continue-on-failure-recursive
+    Failure in user keyword without tag     run_kw=Failure in user keyword without tag
+    Fail   This should be executed
+
+Recursive Continue in test with tag and two nested UK with and without tag
+    [Documentation]    FAIL ${HEADER}\n\n
+    ...    1) kw1a\n\n
+    ...    2) kw1b\n\n
+    ...    3) kw2a\n\n
+    ...    4) kw2b\n\n
+    ...    5) This should be executed
+    [Tags]   robot:continue-on-failure-recursive
+    Failure in user keyword with tag     run_kw=Failure in user keyword without tag
+    Fail   This should be executed
+
+Recursive Continue in test without tag and two nested UK with and without recursive tag
+    [Documentation]    FAIL ${HEADER}\n\n
+    ...    1) kw1a\n\n
+    ...    2) kw1b\n\n
+    ...    3) kw2a\n\n
+    ...    4) kw2b
+    Failure in user keyword with recursive tag     run_kw=Failure in user keyword without tag
+    Fail   This should not be executed
+
+Recursive Continue in test without tag and two nested UK without and with recursive tag
```

Comment:
This name is a bit strange for same reason as the above. Something like "Recursive continue in nested user keyword" could be fine. You probably have a better idea what exactly these two tests are testing and can come up with even better names.